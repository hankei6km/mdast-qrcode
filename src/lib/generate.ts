import QRCode from 'qrcode';
import { MdqrOptions, mdqrOptionsDefaults } from '../qrcode.js';

export async function generateQRCode(
  data: string,
  inOptions: QRCode.QRCodeToDataURLOptions = {
    errorCorrectionLevel: 'M',
    margin: 4,
    scale: 4,
    color: {
      dark: '#000000',
      light: '#ffffff'
    }
  },
  inMdqrOptions: MdqrOptions = {}
) {
  const mdqrOptions = Object.assign({ format: {} }, inMdqrOptions);
  const options = Object.assign({}, inOptions);

  return await QRCode.toDataURL(data, options);
}
