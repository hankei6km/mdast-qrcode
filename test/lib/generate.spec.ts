import { generateQRCode } from '../../src/lib/generate.js';

const consoleErrror = console.error;

afterEach(() => {
  console.error = consoleErrror;
});

describe('generateQRCode()', () => {
  it('should generate QRCode', async () => {
    const res = generateQRCode('test data1');
    expect(await res).toMatchSnapshot();
  });
  it('should generate QRCode with color options', async () => {
    const res = generateQRCode('test data1', {
      color: {
        dark: '#ffffff',
        light: '#000000'
      }
    });
    expect(await res).toMatchSnapshot();
  });
});
