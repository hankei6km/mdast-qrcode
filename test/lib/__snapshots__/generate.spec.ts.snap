// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateQRCode() should generate QRCode 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHQAAAB0CAYAAABUmhYnAAAAAklEQVR4AewaftIAAAKbSURBVO3BQYrkUAwFwXzC979yziy1+mDsKrqEIuJ/rDGKNUqxRinWKMUapVijFGuUYo1SrFGKNUqxRinWKMUapVijFGuUYo1y8VASvkmlS0Kn0iWhU+mS0Kl0SfgmlSeKNUqxRinWKBcvU3lTEk5UvknlTUl4U7FGKdYoxRrl4sOScIfKE0noVD4pCXeofFKxRinWKMUa5eLHJaFTOUlCpzJJsUYp1ijFGuVimCTckYRO5ZcVa5RijVKsUS4+TOWbVLokdCpdEp5Q+UuKNUqxRinWKBcvS8JkSfjLijVKsUYp1igXD6n8MpUTlV9SrFGKNUqxRrl4KAmdyh1J6FS6JHxSEj5J5SQJncoTxRqlWKMUa5T4Hy9KwonKE0noVLoknKh0SehUuiR8ksqbijVKsUYp1igXDyXhjiS8KQmdyptUTpLQqZwk4ZOKNUqxRinWKBcvU7kjCScqdyShU+mScJKETuVNKp9UrFGKNUqxRrl4WRJOVDqVkyR0Kl0SOpU7VLok3KHSJaFTOUlCp/JEsUYp1ijFGiX+xw9LwjepdEnoVLokdCqfVKxRijVKsUa5eCgJ36RyonKShE6lS0Kn8qYkdCpvKtYoxRqlWKNcvEzlTUk4UTlJQqdyonKShDtUTpLQqTxRrFGKNUqxRrn4sCTcoXJHEjqVO5LQqdyh0iWhU+mS0Km8qVijFGuUYo1y8eNUuiQ8kYQTlS4JJ0k4SUKn8kSxRinWKMUa5WIYlTepPKHyTcUapVijFGuUiw9T+UuScKJykoS/rFijFGuUYo1y8bIkfFMSOpU7VLoknKi8KQmdyhPFGqVYoxRrlPgfa4xijVKsUYo1SrFGKdYoxRqlWKMUa5RijVKsUYo1SrFGKdYoxRqlWKP8AyEU+uNLhNAYAAAAAElFTkSuQmCC"`;

exports[`generateQRCode() should generate QRCode with color options 1`] = `"data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAHQAAAB0CAYAAABUmhYnAAAAAklEQVR4AewaftIAAAKYSURBVO3BQarlQAwEwUzh+1+5ZqtVg7H9mC8UIRDWGMUapVijFGuUYo1SrFGKNUqxRinWKMUapVijFGuUYo1SrFGKNUqxRrl4KAm/pNIloVPpktCpdEnoVLok/JLKE8UapVijFGuUi5epvCkJJyq/pPKmJLypWKMUa5RijXLxsSTcofJEEjqVLyXhDpUvFWuUYo1SrFEu/rgkdConSehUJinWKMUapVijXAyThDuS0Kn8ZcUapVijFGuUi4+p/JJKl4ROpUvCEyr/k2KNUqxRijXKxcuSMFkS/mfFGqVYoxRrlIuHVP4ylROVv6RYoxRrlGKNcvFQEjqVO5LQqXRJ+FISvqRykoRO5YlijVKsUYo1ikB4URJOVJ5IQqfSJeFEpUtCp9Il4UsqbyrWKMUapVijXDyUhDuS8KYkdCpvUjlJQqdykoQvFWuUYo1SrFEuXqZyRxJOVO5IQqfSJeEkCZ3Km1S+VKxRijVKsUa5eFkSTlQ6lZMkdCpdEjqVO1S6JNyh0iWhUzlJQqfyRLFGKdYoxRpFIPxhSfgllS4JnUqXhE7lS8UapVijFGuUi4eS8EsqJyonSehUuiR0Km9KQqfypmKNUqxRijXKxctU3pSEE5WTJHQqJyonSbhD5SQJncoTxRqlWKMUa5SLjyXhDpU7ktCp3JGETuUOlS4JnUqXhE7lTcUapVijFGuUiz9OpUvCE0k4UemScJKEkyR0Kk8Ua5RijVKsUS6GUXmTyhMqv1SsUYo1SrFGufiYyv8kCScqJ0n4nxVrlGKNUqxRLl6WhF9KQqdyh0qXhBOVNyWhU3miWKMUa5RijSIQ1hjFGqVYoxRrlGKNUqxRijVKsUYp1ijFGqVYoxRrlGKNUqxRijVKsUb5B8M51x9yYjtMAAAAAElFTkSuQmCC"`;
